include/boost/math/statistics/linear_regression.hpp

```diff
@@ -46,35 +50,39 @@ auto simple_ordinary_least_squares(RandomAccessContainer const & x,
     return std::make_pair(c0, c1);
 }
 
-template<class RandomAccessContainer>
-auto simple_ordinary_least_squares_with_R_squared(RandomAccessContainer const & x,
-                                   RandomAccessContainer const & y)
+template<typename RandomAccessContainer, typename Real = typename RandomAccessContainer::value_type, 
+         typename Size = typename RandomAccessContainer::size_type>
+auto simple_ordinary_least_squares_with_R_squared(RandomAccessContainer const & x, RandomAccessContainer const & y)
+    -> std::tuple<Real, Real, Real>
```

Comment:
If `Real` is a set of integers, then doesn't this break?

Reply:
Like in the event the container passed in is a Matrix? As long is it supports `operator/` and subtraction by a constant than it should behave normally, and if not it would break anyway. `means_and_covariance` returns `std::tuple<Real, Real, Real>` so that should be changed as well if integral types need to return `double`. I know that is the case in almost all of univariate statistics and would be easy enough to change.

Reply:
Ok, I do need to check this then. I had a goal of supporting integer data and if I haven't then that needs to be fixed.

Reply:
I can open another PR to add integer support; that would be easy at this point. Replacing the structured binding in `covariance` with an tuple would drop the required language standard from 17 to 11 as well.